cpp/src/com/google/idea/blaze/cpp/BlazeConfigurationResolver.java

```diff
@@ -98,9 +103,21 @@ public BlazeConfigurationResolverResult update(
         HeaderRootTrimmer.getValidRoots(
             context, blazeProjectData, toolchainLookupMap, targetFilter, executionRootPathResolver);
     builder.setValidHeaderRoots(validHeaderRoots);
+    builder.setTargetToVersionMap(targetToVersion);
     return builder.build();
   }
 
+  @NotNull
+  private static Map<TargetKey, CompilerInfo> getTargetToVersionMap(ImmutableMap<TargetKey, CToolchainIdeInfo> toolchainLookupMap, ImmutableMap<CToolchainIdeInfo, BlazeCompilerSettings> compilerSettings) {
+    Map<ExecutionRootPath, String> compilerVersionByPath =
```

Comment:
Can this function be simplified to something like the following?
```
 return toolchainLookupMap.entrySet().stream()
            .collect(toImmutableMap(e -> e.getKey(), e -> compilerSettings.get(e.getValue()).getCompilerVersion()));
```
or will this be missing something?

Reply:
Also if it is possible, we can have the map key set to the target label string from here instead of transforming it in the BlazeCompileInfoMap class.